### PR TITLE
Doc/restructuring general tutorial

### LINKED ISSUE #510: General-Tutorial-Restructuring
This will split the tutorial into multiple notebooks to make them more bite-sized.

### PR DESCRIPTION
**Pull request by chadcwilliams**
# Description

Restructured the main introductory tutorial by splitting it into four notebooks for easier consumption. I have also edited the notebooks. 

*If the issue is on Github, simply link to it using the '#' symbol; otherwise, provide a notion page link)*:
- resolves #510

# Type of change
- **refactor**: A code change that neither fixes a bug nor adds a feature
- **docs**: Documentation only changes

# Remarks (Optional)
- In each notebook, there are links to each other notebook. I'll need to check the links to make sure they are good.

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -157,7 +157,11 @@ nav:
 - Introduction: 'index.md'
 - Tutorials:
   - Home: 'tutorials/index.md'
-  - Introduction to AutoRA: 'tutorials/Introduction.ipynb'
+  - Basic Tutorial:
+    -I - Components: 'tutorials/basic/Tutorial-I-Components.ipynb'
```
Comment (younesStrittmatter) [inline mkdocs.yml:161]: - I - Components: 'tutorials/basic/Tutorial-I-Components.ipynb'

File: docs/tutorials/basic/Tutorial-I-Components.ipynb [hunk 2]
```diff
@@ -0,0 +1,1212 @@
+{
+  "cells": [
+    {
```
Comment (younesStrittmatter) [inline docs/tutorials/basic/Tutorial-I-Components.ipynb:3]: d

File: mkdocs.yml [hunk 3]
```diff
@@ -157,7 +157,11 @@ nav:
...
+    -II - Loop Constructs: 'tutorials/basic/Tutorial-II-Loop-Constructs.ipynb'
+    -III - Workflow Logic: 'tutorials/basic/Tutorial-III-Workflow-Logic.ipynb'
+    -IV - Customization: 'tutorials/basic/Tutorial-IV-Customization.ipynb'
```
Comment (younesStrittmatter) [inline mkdocs.yml:164]:
```
    - I - Components: 'tutorials/basic/Tutorial-I-Components.ipynb'
    - II - Loop Constructs: 'tutorials/basic/Tutorial-II-Loop-Constructs.ipynb'
    - III - Workflow Logic: 'tutorials/basic/Tutorial-III-Workflow-Logic.ipynb'
    - IV - Customization: 'tutorials/basic/Tutorial-IV-Customization.ipynb'
```

Comment (younesStrittmatter) [inline mkdocs.yml:164]:
```markdown
    - I - Components: 'tutorials/basic/Tutorial-I-Components.ipynb'
    - II - Loop Constructs: 'tutorials/basic/Tutorial-II-Loop-Constructs.ipynb'
    - III - Workflow Logic: 'tutorials/basic/Tutorial-III-Workflow-Logic.ipynb'
    - IV - Customization: 'tutorials/basic/Tutorial-IV-Customization.ipynb'
```

Comment (younesStrittmatter) [inline mkdocs.yml:164]:
```    
- I - Components: 'tutorials/basic/Tutorial-I-Components.ipynb'
```

Comment (younesStrittmatter) [inline mkdocs.yml:164]:
```suggestion
    - I - Components: 'tutorials/basic/Tutorial-I-Components.ipynb'
    - II - Loop Constructs: 'tutorials/basic/Tutorial-II-Loop-Constructs.ipynb'
    - III - Workflow Logic: 'tutorials/basic/Tutorial-III-Workflow-Logic.ipynb'
    - IV - Customization: 'tutorials/basic/Tutorial-IV-Customization.ipynb'
```

**Review by younesStrittmatter — CHANGES_REQUESTED** [review]
building of website fails. This should fix it

**Review by musslick — APPROVED** [review]
Looks great, thanks for the restructuring!

**Review by younesStrittmatter — APPROVED** [review]
looks good

**Review by hollandjg — APPROVED** [review]
Looks great!

**Comment by chadcwilliams** [comment]
Re-implemented with the latest AutoRA versions. However, the latest AutoRA is not yet stable, so this will need to be updated again once AutoRA is stable.

**Pull request merged**